sigstore/_sign.py

```diff
@@ -0,0 +1,114 @@
+import base64
+import hashlib
+
+import jwt
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric import ec
+
+from sigstore._internal.fulcio import (
+    FulcioCertificateSigningRequest,
+    FulcioClient,
+)
+from sigstore._internal.rekor import RekorClient
+from sigstore._internal.oidc import Identity
+
+# from cryptography.x509 import load_pem_x509_certificate
+
+
+REKOR_URL = "https://rekor.sigstore.dev"
+REKOR_API_HEADERS = {"Content-Type": "application/json", "Accept": "application/json"}
+
+
+def _no_output(*a, **kw):
+    pass
+
+
+def sign(file_, identity_token, output=_no_output):
+    """Public API for signing blobs"""
+
+    output(f"Using payload from: {file_.name}")
+    artifact_contents = file_.read().encode()
+    sha256_artifact_hash = hashlib.sha256(artifact_contents).hexdigest()
+
+    output("Generating ephemeral keys...")
+    # Generate the ephemeral private key
+    # private_key = Ed25519PrivateKey.generate()
+    # public_key = private_key.public_key()
+    private_key = ec.generate_private_key(ec.SECP384R1())
+    public_key = private_key.public_key()
+
+    output("Retrieving signed certificate...")
+    # fulcio = FulcioClient("http://localhost:5555")
+    fulcio = FulcioClient()
```

Comment:
I guess we should eventually pipe through some CLI options to control the Fulcio and Rekor instances.

Reply:
Yes, let's make sure to do these as close as possible to how `cosign` does it today.

Reply:
Sure. Let's not block this PR on it.

I've made #27 to track this. We can revisit once we have basic functionality working for signing/verifying since we'll be in a better place to add the CLI flags.